### ISSUE TITLE
onLayout precision error

### ISSUE THREAD

**Issue by pronebird** [Microsoft/reactxp#1056]
Hi,

I've done some testing and it seems that the argument passed into `onLayout` function is rounded down. At the same time, `UserInterface` measuring functions return the fractional values.

```
  private onLayout = async (layout: Types.ViewOnLayoutEvent) => {
    const layout2 = await UserInterface.measureLayoutRelativeToAncestor(this.measuringView.current!, this);
    console.info('layout1 is ', layout);
    console.info('layout2 is ', layout2);
  };
};
```

The result is:

```
layout1 is  {x: 0, y: 0, width: 101, height: 26}
layout2 is  {x: 0, y: 0, width: 101.125, height: 26}
```

**Comment by erictraut** [comment]
Which platform? One the web, we implement these functions using calls to the browser. On native platforms, onLayout and the measurement functions pass through almost directly to RN.

**Comment by pronebird** [comment]
Sorry. It's the web platform. I see that `onLayout` and `measureLayoutRelativeToAncestor` return different values, so I assume something "rounds" the numbers along the way in the case of `onLayout`.

**Comment by erictraut** [comment]
I don't see any place in the ReactXP code where we're rounding these values. 

web/ViewBase.tsx is using the offsetWidth and offsetHeight properties of the DOM node to calculate the dimensions that are returned to the onLayout callback.

web/UserInterface.tsx is using the method call getBoundingClientRect(). 

I would expect the two to return the same value. Maybe we should use getBoundingClientRect for both cases.

**Comment by pronebird** [comment]
Yeah, I'd expect the same values too. Weird thing. I measure the view with the `Text` node inside. I think that `getBoundingClientRect` may be returning the fractional values then.

**Comment by erictraut** [comment]
This might be browser-specific behavior too. Have you tried on other browsers out of curiosity?

**Comment by pronebird** [comment]
I think you're right. I haven't tried on other browsers because we use Electron, but I can perhaps find a boilerplate for RX and run it online. I'll let you know.

**Comment by fbartho** [comment]
@pronebird -- is this still a concern? I'd love to fix this if possible. I haven't been using those measurement APIs on web much yet.
